pkg/k8s/scanner/scanner.go

```diff
@@ -173,22 +166,43 @@ func (s *Scanner) scanVulns(ctx context.Context, artifact *artifacts.Artifact, o
 	return resources, nil
 }
 
-func (s *Scanner) scanMisconfigs(ctx context.Context, artifact *artifacts.Artifact) (report.Resource, error) {
-	configFile, err := createTempFile(artifact)
+func (s *Scanner) scanMisconfigs(ctx context.Context, k8sArtifacts []*artifacts.Artifact) ([]report.Resource, error) {
+	folder, artifactsByFilename, err := generateTempFolder(k8sArtifacts)
 	if err != nil {
-		return report.Resource{}, xerrors.Errorf("scan error: %w", err)
+		return nil, xerrors.Errorf("failed to generate temp folder: %w", err)
 	}
 
-	s.opts.Target = configFile
+	s.opts.Target = folder
 
 	configReport, err := s.runner.ScanFilesystem(ctx, s.opts)
-	// remove config file after scanning
-	removeFile(configFile)
+	// remove config files after scanning
+	removeFolder(folder)
+
 	if err != nil {
-		return report.CreateResource(artifact, configReport, err), err
+		return nil, xerrors.Errorf("failed to scan filesystem: %w", err)
+	}
+	resources := make([]report.Resource, 0, len(k8sArtifacts))
```

Comment:
Is there any benefit of defining this of fixed length? Why not simplify like so

```suggestion
	var resources []report.Resource
```

Reply:
try to avoid a few copy operations and memory allocations by grabbing it all up front. I'm not sure about (it's a quota  )).

do you think we should use a simple construction?

Reply:
I haven't benchmarked this so I can't say for sure but I would assume the Go runtime can grow the slice as needed without much of an overhead.

Reply:
`make` is faster than `var`, so if we know the length beforehand, we should use `make` for performance reasons.
https://sampath04.hashnode.dev/make-your-go-code-efficient-using-make-when-creating-slices

If the length of the slice is small enough, I personally prefer `var` because it's easier to read. In this case, k8s resources can be a lot. It makes sense to use `make`.